### DISCUSSION TITLE
not able to pass custom struct to bt file

### DISCUSSION THREAD

**Discussion by Engg-blogger** [bpftrace/bpftrace#3471] [General]
Hi Team - I was looking at the issue : https://github.com/bpftrace/bpftrace/issues/999. And i was trying to create a struct to read the value of filename from `tracepoint: 'tracepoint:sched:sched_process_exit'.`

```
#include <linux/types.h>
struct data 
{
  long pad;
  int filename;
  pid_t pid;
  pid_t old_pid;
}
tracepoint:sched:sched_process_exit {
	$data = (struct data*)args;
        printf("%s,%d,%s\n", strftime("%D %H:%M:%S", nsecs), pid, str((uint64)args + (uint64)($data->filename & 0xFFFF)));
}
```
But, I am getting below error:

```
ERROR: Cannot cast from struct type "struct _tracepoint_sched_sched_process_exit"
    $data = (struct data*)args;
            ~~~~~~~~~~~~~~~~
ERROR: Cannot cast from struct type "struct _tracepoint_sched_sched_process_exit"
        printf("%s,%d,%s\n", strftime("%D %H:%M:%S", nsecs), pid, str((uint64)args + (uint64)($data->filename & 0xFFFF)));
```
Could you advise?

**Comment by viktormalik** [comment]
Why do you want to do the cast? You should be able to access the argument directly by `args.filename`.

**Reply by Engg-blogger** [reply]
I am facing the same issue as mentioned in the link i shared above. When using args.filename, i am getting below error:
ERROR: Struct/union of type 'struct _tracepoint_sched_sched_process_exit' does not contain a field named 'filename'

**Reply by viktormalik** [reply]
Well, that's because you're trying to attach to `tracepoint:sched:sched_process_exit` which doesn't have the `filename` argument:
```
# bpftrace -lv 'tracepoint:sched:sched_process_exit'
tracepoint:sched:sched_process_exit
    char comm[16]
    pid_t pid
    int prio
```
Note that #999 is about attaching to `tracepoint:sched:sched_process_exec`, i.e. a different tracepoint.
